packages/untp-test-suite/src/core/types/index.ts

```diff
@@ -25,4 +25,37 @@ export interface TestErrors {
   errors: ErrorObject[] | ICredentialConfigError[] | null;
 }
 
-export interface TestSuiteResult extends ConfigContent, TestErrors {}
+export interface IValidatedCredentials extends ConfigContent, TestErrors {}
+
+export interface ICredentialTestResult {
+  credentialType: string;
+  version: string;
+  path: string;
+  result: string;
+  warnings?: string[];
+  error?: string[];
+}
+
+export interface IFinalReport {
+  finalStatus: FinalStatus;
+  finalMessage: TestSuiteMessage;
+}
+
+export interface TestSuiteResult extends IFinalReport {
+  credentials: ICredentialTestResult[];
+}
+
+export type FinalStatus = 'FAILED' | 'PASS' | 'WARN';
+
+export enum TemplateName {
+  failed = 'FAILED',
+  pass = 'PASS',
+  warn = `WARN`,
+  finalReport = 'finalReport',
+}
```

Comment:
We have created four template files: `error.hbs`, `pass.hbs`, `warning.hbs`, and `finalReport.hbs`. You can view the folder at `untp-test-suite/src/templates/templateMessages` for more detail. If you use this type enum here to load the exact template file, I think the enum values should be the same as the template file name.

You can refer to this:

```ts
export enum TemplateName {
  error = 'error',
  pass = 'pass',
  warning = `warning`,
  finalReport = 'finalReport',
}
```